presto-tests/src/test/java/io/prestosql/execution/sessionpropertymanagers/TestDbSessionPropertyManagerIntegration.java

```diff
@@ -0,0 +1,251 @@
+                .put("session-property-manager.db.password", MYSQL_TEST_PASSWORD)
+                .build();
+
+        setDbSessionPropertyConfigurationManager(configs);
+        SessionPropertiesDao dao = getTestingMySqlDao(configs);
+
+        dao.dropSessionPropertiesTable();
+        dao.dropSessionClientTagsTable();
+        dao.dropSessionSpecsTable();
+        dao.createSessionSpecsTable();
+        dao.createSessionClientTagsTable();
+        dao.createSessionPropertiesTable();
+
+        // Configure the session property for users with user regex user1.*
+        dao.insertSpecRow(1, "user1.*", null, null, null, 0);
+        dao.insertSessionProperty(1, EXAMPLE_PROPERTY, EXAMPLE_VALUE_CONFIGURED.toString());
+
+        // All queries with matching session should have overridden session properties
+        assertSessionPropertyValue("user123", EXAMPLE_VALUE_CONFIGURED);
+
+        // Add a spec and simulate a bad database operation to intentionally fail further queries
+        dao.insertSpecRow(2, "user3.*", null, null, null, 0);
+        dao.insertSessionProperty(2, EXAMPLE_PROPERTY, EXAMPLE_VALUE_CONFIGURED.toString());
+        dao.dropSessionPropertiesTable();
+
+        // Reloading should fail now, old values should still be in use.
+        assertSessionPropertyValue("user123", EXAMPLE_VALUE_CONFIGURED);
+        assertSessionPropertyValue("user345", EXAMPLE_VALUE_DEFAULT);
+
+        // Fix the database by re-constructing the dropped table
+        dao.createSessionPropertiesTable();
+        dao.insertSessionProperty(1, EXAMPLE_PROPERTY, EXAMPLE_VALUE_CONFIGURED.toString());
+        dao.insertSessionProperty(2, EXAMPLE_PROPERTY, EXAMPLE_VALUE_CONFIGURED.toString());
+
+        // Fixing the database should enable successful reloading again.
+        assertSessionPropertyValue("user123", EXAMPLE_VALUE_CONFIGURED);
+        assertSessionPropertyValue("user312", EXAMPLE_VALUE_CONFIGURED);
+    }
+
+    private void assertSessionPropertyValue(String user, Duration expectedValue)
+    {
+        Session session = testSessionBuilder()
+                .setIdentity(new Identity(user, Optional.empty()))
+                .build();
+
+        MaterializedResult result = queryRunner.execute(session, "SHOW SESSION");
+        String actualValueString = (String) result
+                .getMaterializedRows()
+                .stream().filter(row -> (row.getField(0).equals(EXAMPLE_PROPERTY)))
+                .collect(Collectors.toList())
+                .get(0)
+                .getField(1);
+
+        assertEquals(Duration.valueOf(actualValueString), expectedValue);
+    }
+
+    private void setDbSessionPropertyConfigurationManager(Map<String, String> configs)
+    {
+        SessionPropertyDefaults sessionDefaults = queryRunner.getCoordinator().getSessionPropertyDefaults();
+        sessionDefaults.setConfigurationManager("db-test", configs);
+    }
+
+    private SessionPropertiesDao getTestingMySqlDao(Map<String, String> configs)
```

Comment:
Rather than passing the configs as a map, I would inline this method. We actually only need the URL, since the testing MySQL server puts the credentials in the URL.